### PR TITLE
docs: revamp the Mac OS section.

### PR DESCRIPTION
**Pull request by matz-e**
Don't provide any sysconfig, Homebrew is a moving target, and an independent Spack installation should be much more reliable.

Backports a Spack PR to avoid issues with M1/MacOS definitions across all software.

### CODE REVIEW

File: bluebrain/documentation/setup_personal.md [hunk 1]
```diff
@@ -2,22 +2,46 @@
...
-Install software on OS X, using Homebrew for binary packages.
+To install end-user software on OS X, please defer to `brew`.
 
-On OS X the build process is very similar to Ubuntu. To avoid building
-the whole stack from source we can likewise use another package manager to provide precompiled binaries.
-To that end we have successfully used Homebrew. We also provide a skeleton 
-`packages.yaml` that you should review and adapt to your needs.
+Before starting, please install `brew` and make sure that there is a
+working Python on your machine, preferably from XCode or another *stable*
+source.
```
Comment (pramodk) [inline bluebrain/documentation/setup_personal.md]: Just about this - I prefer this as well but in the past have seen (undocumented, minor) issues when using system python. Your didn't see anything major on your Apple M1 test?

Comment (matz-e) [reply bluebrain/documentation/setup_personal.md]: I used that Python to bootstrap a Spack one 🤷

Comment (matz-e) [reply bluebrain/documentation/setup_personal.md]: FWIW, building Python itself was faster than building CMake…

Comment (pramodk) [reply bluebrain/documentation/setup_personal.md]: > I used that Python to bootstrap a Spack one

Ah ok 👍

File: lib/spack/spack/build_systems/autotools.py [hunk 2]
```diff
@@ -415,6 +417,13 @@ def configure(self, spec, prefix):
         with working_dir(self.build_directory, create=True):
             inspect.getmodule(self).configure(*options)
 
+    def setup_build_environment(self, env):
+        if (self.spec.platform == 'darwin'
+                and macos_version() >= Version('11')):
+            # Many configure files rely on matching '10.*' for macOS version
+            # detection and fail to add flags if it shows as version 11.
+            env.set('MACOSX_DEPLOYMENT_TARGET', '10.16')
```
Comment (alexsavulescu) [inline lib/spack/spack/build_systems/autotools.py:425]: Just curious, why `10.16` ? 
For NEURON we set `10.9`

Comment (matz-e) [reply lib/spack/spack/build_systems/autotools.py:425]: That's from upstream, I didn't really find documentation on this.

Comment (alexsavulescu) [reply lib/spack/spack/build_systems/autotools.py:425]: Ok, looks like it's for config purposes only.

**Review by alexsavulescu — APPROVED** [review]
LGTM. Was playing around with `spack externals find` a couple of months ago and that didn't yield that many libs. Will play again with it soon ™️

**Comment by matz-e** [comment]
> LGTM. Was playing around with `spack externals find` a couple of months ago and that didn't yield that many libs. Will play again with it soon ™️

It's not super useful, but I also don't want `brew` to pull the rug out from underneath Spack. So the current limitation feels OK on MacOS.

Thanks for the review!

**Pull request merged**